### DISCUSSION TITLE
Advanced filtering of the plugin directory.

### DISCUSSION THREAD

**Discussion by alfredbuehler** [aljawaid/PluginManager#52] [Ideas]
![image](https://github.com/aljawaid/PluginManager/assets/2079289/8497c889-f0ff-46d0-a05c-9e2ede19fdf4)

It would be great, if there were a possibility to filter on plugin types. BONUS: Plugins with updates.

**Comment by aljawaid** [comment]
I totally agree and related to #42 but after trying many many times, I don't know how.  It seems to be the way the code is, a table for each plugin so a major re-write of the default code will be needed I think.

I do plan to move all the svg code out of the templates, something I am doing with ContentCleaner which is making the code easier for me to update and upgrade. But again, it's an ongoing work in progress.

I'd love to hear your ideas if you have any?

**Reply by alfredbuehler** [reply]
Maybe I can help.

**Reply by aljawaid** [reply]
Sure you're more than welcome to.

Previously, I struggled with getting a count of updates so that I could at least show the user they have x updates available.

The filtering was very hard for me to make it work. Because each plugin is ONE table.

The JS I have used for hovering over inputs is just to make it faster and easier for keyboard users

I thought about changing the whole layout but    that again would get too complicated.

**Reply by alfredbuehler** [reply]
Created an issue #53 for that feature/enhancement.

**Reply by alfredbuehler** [reply]
It's done, and I really like it! :wink:

**Reply by aljawaid** [reply]
@alfredbuehler I am getting this error in the code scan for your part of the js code, is it safe to ignore or do you need to action it?

![image](https://github.com/aljawaid/PluginManager/assets/10233708/beb86ce3-5843-49a1-b741-0acb922901bb)

https://github.com/aljawaid/PluginManager/blob/7feffa60db8859b0e85fc97b9a105c39b6b0dd0c/Assets/js/plugin-manager.js#L7-L11

**Reply by alfredbuehler** [reply]
The const keyword was introduced in ES6 (2015. Maybe your scanner uses an older standard? I think I fixed it.
God bless your code checker!
I didn't get an error when I checked in… ?

**Reply by aljawaid** [reply]
I'm new to code linting, I am slowly working on the erors and ignoring those which would not (hopefully) break code, e.g. line lengths etc.

**Reply by alfredbuehler** [reply]
Sorry, I don't have the time for such things.

**Reply by aljawaid** [reply]
> Sorry, I don't have the time for such things.

errrm you dont need to do anything... I was just telling you :)

Anyway, this error has been removed thanks to your fix, I will gradually work on the rest of the files.  Thanks for this feature.

**Reply by alfredbuehler** [reply]
You're right, I was somewhat excited. Please accept my apology.

**Reply by aljawaid** [reply]
no no it is fine, don't worry.  Anyhow, I have finished all the tests and all the errors are gone now, so we now have cleaner code.

I am going to migrate the svg code and then I will do the next release. 👍🏼

**Reply by alfredbuehler** [reply]
I'm linting now locally, to avoid any trouble for my future submissions.

**Reply by alfredbuehler** [reply]
@aljawaid You should provide tools/config to lint code locally, BEFORE committing into master. It's annoying to get slapped just because of a minor overseen style issue. My 2 cents/proposal.

**Reply by aljawaid** [reply]
I'm currently only doing all this in my spare time and don't have the local resources as such.  Don't worry about the linting if it is styling related, as long as the code works, its all good.

It's not meant to be a slap, please don't take it that way, for a non-developer like me, its a learning curve.
